tower-http/src/compression/compression_filter.rs

```diff
@@ -0,0 +1,26 @@
+use crate::compression_utils::default_compression_filter_predicate;
+
+/// A filter which any response Parts needs to pass to be compressed
+pub trait CompressionFilter: Copy {
```

Comment:
I think requiring `Copy` might be too limiting in some cases. For examples users might want to implement `CompressionFilter` for something like

```rust
struct MyFilter
    compression_enabled: Arc<AtomicBool>,
}
```

This isn't possible since `Arc` isn't `Copy` but only `Clone`.

So I think we should require `CompressionFilter: Clone` instead.

What do you think?

Reply:
Also wondering if `CompressionPredicate` would be a better name? Tower already has [`Filter`](https://docs.rs/tower/0.4.10/tower/filter/struct.Filter.html) which is something very different 🤔